syntax = "proto3";

package inference;

option go_package = "webrtc-pion-face-engine/proto;pb";

service FaceDetection {
  rpc DetectFaces (FrameRequest) returns (DetectionResponse);
}

message FrameRequest {
  // Moved metadata to the beginning of the message
  int32 width = 1;
  int32 height = 2;
  int32 channels = 3; // Could indicate encoded vs raw
  
  // Frame metadata
  int64 timestamp = 4;
  string frame_id = 5;
  string codec = 6;     // "VP8" - tells C++ what to decode
  
  // Large data payload moved to the end
  bytes encoded_frame = 7;    // VP8 encoded data
}

message BoundingBox {
  int32 x = 1;
  int32 y = 2;
  int32 width = 3;
  int32 height = 4;
  float confidence = 5;
}

message DetectionResponse {
  repeated BoundingBox faces = 1;
  int64 timestamp = 2;
  string frame_id = 3;
  int64 processing_time_ms = 4;
}
